### ISSUE TITLE
Move topics from one community to another.

### ISSUE THREAD

**Issue by Tealk** [LemmyNet/lemmy#2345]
For front end issues, use [lemmy-ui](https://github.com/LemmyNet/lemmy-ui)

### Is your proposal related to a problem?

There are a lot of posts from topic xy and that's why you create a community for topic xy; it would be handy if you could then move the posts.

### Describe the solution you'd like

Move topics from one community to another.

### Describe alternatives you've considered

I have re-posted posts, but that only makes sense if you have written them yourself, otherwise the creator is lost.

### Additional context

nothing

**Comment by dessalines** [comment]
What's wrong with the cross-post button? The creator isn't lost, it both links to the original post, and you will be seen as the post creator.

**Comment by sam365724** [comment]
This is an important functionality for mod's, its help communities to stay focussed and clean when user post into wrong communities.

**Comment by dessalines** [comment]
I don't see much value in this since you can just use the cross-post button, and all the original content gets copied, while preserving the original post. Moving rather than copying posts could get pretty complicated with mods, federated communities, etc. If the only reason to add this is to preserve the original creator, then we could add that to the cross-post blurb that gets copied over with text posts.

**Comment by sam365724** [comment]
Is crosspost implemented yet? I can't seem to find that button.

**Comment by dessalines** [comment]
This button:

![IMG_20220829_143724](https://user-images.githubusercontent.com/930215/187273816-39330efe-fd79-4356-b716-e12b8d946053.jpg)

**Comment by Nutomic** [comment]
I agree that this is an important feature, mainly on instances that are limited to a couple of fixed, theme-specific communities such as feddit.it. It doesnt matter much on an instance like lemmy.ml, where anyone can create communities, and in practice there are few rules on what can be posted where.

Crosspost doesnt really solve the problem, because if the original community should be clean without offtopic posts, then the original post would have to be deleted, and all the comments (and votes) would disappear with it.

Implementing this seems difficult though. Not only would things have to be rewritten in the local database, this action would also have to be federated to other instances. And then there might be cases where a topic needs to be moved from a community on instance alpha to a community on beta, which seems very complicated.

So I think this would be a worthwile feature to have, but much too complicated to implement now, when there are so many other features which are easier to implement and will have a higher impact.

**Comment by sam365724** [comment]
If you need to do it, you can use this SQL query - it's straight forward, the ids at the end are one or multiple posts...

`UPDATE post set community_id=24 where id IN (89, 84, 65);`

**Comment by dessalines** [comment]
Every time we use the phrase "moving", we should remember that it really means "rewriting history". Moving a post from one community to another would entail copying and re-creating all the comments, votes, and re-federating them.

**Comment by dessalines** [comment]
Closing this as it would require re-writing history, without the consent of any of the commenters.

**Comment by Nutomic** [comment]
NodeBB is talking about implementing this with a `Move` activity. In this case there would be no need to recreate anything.

https://activitypub.space/topic/58/moving-topics-contexts-between-communities
